File: examples/SDK/dotnet/WorkingWithItems/BatchWriteItem.cs
using System;
using System.Threading.Tasks;

using Amazon;
using Amazon.DynamoDBv2;
using Amazon.DynamoDBv2.Model;
using System.Collections.Generic;

namespace DotnetSamples.WorkingWithItems
{

    public class BatchWriteItem
    {
        /// <summary>
        /// Example that writes to a DynamoDb more than one item in one call.
        /// </summary>
        /// <returns></returns>
        public static async Task BatchWriteExampleAsync()
        {
            //Create Client
            var client = new AmazonDynamoDBClient(RegionEndpoint.USWest2);

            //Definition of the first item to put on a table
            var putFirstItem = new PutRequest(new Dictionary<string, AttributeValue>
                                        {
                                            {"pk", new AttributeValue("jim.bob@somewhere.com")},
                                            {"sk", new AttributeValue("metadata")},
                                            {"attribute1",new AttributeValue("Attribute1 value")  }
                                            //Add other attributes as you need 
                                        }
                                        );
            //Definition of the second item to put on a table
            var putSecondItem = new PutRequest(new Dictionary<string, AttributeValue>
                                        {
                                            {"pk", new AttributeValue("jose.schneller@somewhere.com")},
                                            {"sk", new AttributeValue("metadata")},
                                            {"attribute1",new AttributeValue("Attribute1 value")  }
                                            //Add other attributes as you need 
                                        }
                                       );
            //Definition of an item to delete
            var deleteItem = new DeleteRequest(new Dictionary<string, AttributeValue>
                                        {
                                            {"pk", new AttributeValue("vikram.johnson@somewhere.com")},
                                            {"sk", new AttributeValue("metadata")},
                                        }
                                        );

            //Request that group all the previous Put & Delete actions
            var writeRequest = new BatchWriteItemRequest
            {
                RequestItems = new Dictionary<string, List<WriteRequest>>
                { 
                    { 
                        //Name of the table
                        "RetailDatabase",
                        new List<WriteRequest>()
                            {
                                new WriteRequest(putFirstItem),
                                new WriteRequest(putSecondItem),
                                new WriteRequest(deleteItem)
                            }
                    },//You can execute other collections of requests on other tables at the same time
                    {
                        //Name of the table
                        "RetailDatabase2",
                        new List<WriteRequest>()
                            {
                                new WriteRequest(putFirstItem),
                                new WriteRequest(putSecondItem),
                                new WriteRequest(deleteItem)
                            }
                    }
                }
            };

            try
            {
                //Execution of the request
                var responseWrite = await client.BatchWriteItemAsync(writeRequest);
                
                Console.WriteLine($"Status {responseWrite.HttpStatusCode}");
                Console.WriteLine($"Number of items not processed that you need to try again:{responseWrite.UnprocessedItems}");
            }
            catch (Exception e)
            {
                Console.Error.WriteLine(e.Message);
            }

        }


    }
}

File: examples/SDK/golang/WorkingWithItems/batchWriteItem.go
package main

import (
	"fmt"

	"github.com/aws/aws-sdk-go/aws"
	"github.com/aws/aws-sdk-go/aws/session"
	"github.com/aws/aws-sdk-go/service/dynamodb"
)


func main() {

	// Create Session
	sess := session.Must(session.NewSessionWithOptions(session.Options{
		SharedConfigState: session.SharedConfigEnable,
	}))

	// Create DynamoDB Client
	svc := dynamodb.New(sess, aws.NewConfig())

	// Write Items
	input := &dynamodb.BatchWriteItemInput{
		RequestItems: map[string][]*dynamodb.WriteRequest{
			"RetailDatabase": {
				{
					PutRequest: &dynamodb.PutRequest{
						Item: map[string]*dynamodb.AttributeValue{
							"pk": {
								S: aws.String("jose.schneller@somewhere.com"),
							},
							"sk": {
								S: aws.String("metadata"),
							},
							"firstName": {
								S: aws.String("jose"),
							},
							"lastName": {
								S: aws.String("schneller"),
							},
							"name": {
								S: aws.String("Jose Schneller"),
							},
							"username": {
								S: aws.String("joses"),
							},
						},
					},
				},
				{
					PutRequest: &dynamodb.PutRequest{
						Item: map[string]*dynamodb.AttributeValue{
							"pk": {
								S: aws.String("jim.bob@somewhere.com"),
							},
							"sk": {
								S: aws.String("metadata"),
							},
							"firstName": {
								S: aws.String("jim"),
							},
							"lastName": {
								S: aws.String("bob"),
							},
							"name": {
								S: aws.String("Jim Bob"),
							},
							"username": {
								S: aws.String("jbob"),
							},
						},
					},
				},
				
			},
		},
	}


	result, err := svc.BatchWriteItem(input)

	// Catch Error
	if err != nil {
		fmt.Println("BatchWrite API call failed:")
		fmt.Println((err.Error()))
	}

	fmt.Println("Successfully added item to table")
	fmt.Println(result)
}
